hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/renderers/AbstractGridButtonConverter.java

```
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.customrenderers.renderers;
+
+import java.io.Serializable;
+import java.util.Locale;
+
+import org.eclipse.hawkbit.ui.customrenderers.client.renderers.FontIconData;
+import org.eclipse.hawkbit.ui.rollout.StatusFontIcon;
+
+import com.vaadin.data.util.converter.Converter;
+
+/**
+ * Converter that adapts to a model and converts to a grid-button presentation.
+ *
+ * @param <T>
+ *            The type the converter adapts to
+ */
+public abstract class AbstractGridButtonConverter<T> implements Converter<FontIconData, T> {
+
+    private static final long serialVersionUID = 1L;
+
+    private GridButtonAdapter<T> adapter;
+
+    @Override
+    public T convertToModel(final FontIconData meta, final Class<? extends T> targetType, final Locale locale) {
+        // not needed
+        return null;
```

Should throw a `UnsupportedOperationException`